plugins/woocommerce/src/Blocks/Templates/ProductCatalogTemplate.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+use Automattic\WooCommerce\Blocks\Utils\BlockTemplateUtils;
+
+/**
+ * ProductCatalogTemplate class.
+ *
+ * @internal
+ */
+class ProductCatalogTemplate extends AbstractTemplate {
+
+	/**
+	 * The slug of the template.
+	 *
+	 * @var string
+	 */
+	const SLUG = 'archive-product';
+
+	/**
+	 * The title of the template.
+	 *
+	 * @var string
+	 */
+	public $template_title;
+
+	/**
+	 * The description of the template.
+	 *
+	 * @var string
+	 */
+	public $template_description;
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		$this->template_title       = _x( 'Product Catalog', 'Template name', 'woocommerce' );
+		$this->template_description = __( 'Displays your products.', 'woocommerce' );
+
+		add_action( 'template_redirect', array( $this, 'render_block_template' ) );
+		add_filter( 'post_type_archive_title', array( $this, 'update_product_archive_title' ), 10, 2 );
+	}
+
+	/**
+	 * Renders the default block template from Woo Blocks if no theme templates exist.
+	 */
+	public function render_block_template() {
```

Comment:
Can we not use this `class` more generically to render the other product archive templates and remove the additional classes?

Reply:
I guess it could be done, but what would be the benefit of doing so?

One of the benefits I see from having one file per template is that there is a clear relation between:

* The template PHP file where we define title, description, specific hooks...
* Templates stored in the database.
* Template files from themes.

All of them sharing the same slug. While we could move the logic from other archive templates into this file, that would break the 1:1 relation between the above, which IMO would make things more confusing. :thinking:

Reply:
I guess the change only makes sense if/once we remove the other product archive templates altogether and render the Product Catalog template in every case _unless_ the user has added a specific template via the Add New UI in the Site Editor.